SNIPS/snip-15.md

```diff
@@ -0,0 +1,40 @@
+# Draft---
+snip: 15
+title: Only an account can call its `execute`
+author: Ilia Volokh <iliav@starkware.co>
+discussions-to: 
+status: Draft
+type: Core
+created: 2024-01-08
+---
+
+## Simple Summary
+Restrict Starknet so only an account can call its own `execute` function, preventing internal calls from other contracts.
+
+## Abstract
+
+This proposal defines a protocol-level restriction that will be imposed on callers to the `execute` entrypoint of an account. The restriction requires the caller address to equal the address of the account of the `execute` in question.
+
+## Motivation
+
+Users act through their accounts via the `execute` entrypoint. Being able to access the execute of someone's account is tantamount to acting on their behalf. The default behavior should defend naive users and developers against such vulnerabilities.
+
+On the other hand, the current status quo is that every account's `execute` explicitly contains a line which enforces the call is not internal. Accounts without this line are exposed.
```

Comment:
> is that every account's `execute` explicitly contains a line which enforces the call is not internal

It's not true that every account contains it. It'd would be clearer to say "all safe account implementations explicitly contain ... ".